### ISSUE TITLE
Investigate removing argumentBindingException on HubMethodInvocationMessage

### ISSUE THREAD

**Issue by JamesNK** [aspnet/SignalR#1941]
https://github.com/aspnet/SignalR/pull/1931#discussion_r180485370

Was originally added because of https://github.com/aspnet/SignalR/issues/815

**Comment by davidfowl** [comment]
Ok I took a quick look at this and the key here is that the error was associated with the actual invocation id.

**Comment by moozzyk** [comment]
I guess this requires deciding whether the connection should be closed or not in these scenarios. The original problem (#815) was that we were able to return an invocation error for an invalid/non-existing hub method but as soon as the method name was valid but there was something wrong with parameters we were not able to handle this and closed the connection hard. It was super weird. Now, that SignalR contains a Close message I think it may  be OK to close the connection with an error in both cases. It could/would actually help developers find their errors sooner rather than later because it is never valid to invoke a non-existing hub method from the client or to not provide valid parameters (regardless whether this is arity and/or types).

**Comment by davidfowl** [comment]
We agreed not to close the connection but instead we'd create another pseudo message type on the server that has an error and invocation id.

**Comment by moozzyk** [comment]
Yeah, I did consider this when fixing #815 but it felt wrong to abuse messages for the internal error handling because they are coupled to the protocol.

**Comment by davidfowl** [comment]
It’s a but cleaner than what is there today. Especially since invocations are used for outgoing sends.

**Comment by moozzyk** [comment]
I can see that.
